V2/api/evmToEVMSwap_setToAmount/src/index.ts

```diff
@@ -0,0 +1,253 @@
+import { ethers } from "ethers";
+import axios from "axios";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Load environment variables from .env file
+const privateKey: string = process.env.PRIVATE_KEY!;
+const integratorId: string = process.env.INTEGRATOR_ID!;
+const FROM_CHAIN_RPC: string = process.env.FROM_CHAIN_RPC_ENDPOINT!;
+
+// Define chain and token addresses
+const fromChainId = "56"; // BNB
+const toChainId = "42161"; // Arbitrum
+const fromToken = "0x55d398326f99059fF775485246999027B3197955"; // USDT
+const toToken = "0xaf88d065e77c8cC2239327C5EDb3A432268e5831"; // USDC
+
+// Set up JSON RPC provider and signer 
+const provider = new ethers.providers.JsonRpcProvider(FROM_CHAIN_RPC);
+const signer = new ethers.Wallet(privateKey, provider);
+
+// Function to get token information from Squid API
+const getTokens = async () => {
+  try {
+    const result = await axios.get('https://apiplus.squidrouter.com/v2/sdk-info', {
+      headers: {
+        'x-integrator-id': integratorId,
+      },
+    });
+    return result.data.tokens;
+  } catch (error) {
+    console.error("Error fetching token data:", error);
+    return [];
+  }
+};
+
+// Function to find a specific token in the token list
+function findToken(tokens: any[], address: string, chainId: string) {
+  if (!Array.isArray(tokens)) {
+    console.error("Invalid tokens data structure");
+    return null;
+  }
+
+  return tokens.find(t => 
+    t.address.toLowerCase() === address.toLowerCase() && 
+    t.chainId === chainId
+  );
+}
+
+// Function to calculate the amount of fromToken needed based on user-specified toAmount
+function calculateFromAmountBasedOnToAmount(toAmount: string, fromToken: any, toToken: any): string {
+  const fromTokenDecimals = fromToken.decimals;
+  const toTokenDecimals = toToken.decimals;
+  const fromTokenUsdPrice = fromToken.usdPrice;
+  const toTokenUsdPrice = toToken.usdPrice;
+
+  // Convert toAmount to a number, considering its decimals
+  const toAmountNumber = Number(toAmount) / (10 ** toTokenDecimals);
+
+  // Calculate the USD value of the toAmount
+  const usdValue = toAmountNumber * toTokenUsdPrice;
+
+  // Calculate the amount of fromToken needed to match this USD value
+  const fromAmountNumber = usdValue / fromTokenUsdPrice;
+
+  // Add a 0.5% buffer for price fluctuations and fees
```

Comment:
Would probably bump this to 1% just in case for larger transactions

Reply:
Will do